src/verkle/keyvals/mod.rs

```diff
@@ -0,0 +1,50 @@
+// https://github.com/gballet/verkle-block-sample
```

Comment:
???

Reply:
This part of code was taken from @gballet repo. So the idea was do not change this code

Reply:
Thanks for the review, I have integrated some of the changes in `master`. I can also make the types you are interested in public, so that you don't have to duplicate the code.

One question, for my personal edification @leviska : how would you propagate the error in 

```rust
        let (keys, values): (Vec<[u8; 32]>, Vec<Option<[u8; 32]>>) = rlp
            .iter()
            .map(|r| r.as_val::<Tuple>()?.try_into().unwrap())
            .unzip();
```

?

Reply:
If you make these types public, it will help a lot!
But I suppose there's some issues about proof format decoding in go and rust, aren't they?

Reply:
There's two ways I can recommend to do it, both are available in this [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a4f64425c7fcf61fd6c21a2c58b72445)

First is to collect into `Result`, which will iterate while Ok, and stop on first err. You can find more common code examples [here](https://doc.rust-lang.org/rust-by-example/error/iter_result.html)

```rust
fn simple(rlp: Vec<u8>) -> Result<(Vec<u8>, Vec<Option<u8>>), ()> {
    let result: Result<Vec<_>, _> = rlp.iter().map(|r| as_val(*r)).collect();
    let (keys, values): (Vec<u8>, Vec<Option<u8>>) = result?.into_iter().unzip();
    Ok((keys, values))
}
```

The problem with this approach, that in order to unzip, we need to collect values into a buffer, which can be fine, but still it's unnecessary work, so there is a better way using `itertools`:

```rust 
fn no_buffer(rlp: Vec<u8>) -> Result<(Vec<u8>, Vec<Option<u8>>), ()> {
    let result = itertools::process_results(rlp.iter().map(|r| as_val(*r)), |iter| iter.unzip());
    let (keys, values): (Vec<u8>, Vec<Option<u8>>) = result?;
    Ok((keys, values))
}

// or even in one line
fn no_buffer_oneliner(rlp: Vec<u8>) -> Result<(Vec<u8>, Vec<Option<u8>>), ()> {
    itertools::process_results(rlp.iter().map(|r| as_val(*r)), |iter| iter.unzip())
}
```